llvm/0202-hack-add-system-search-paths.patch

```diff
@@ -0,0 +1,27 @@
+From 7e9b0669061b7d3588fdfe18dc31a780e5ada69d Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Mateusz=20Miku=C5=82a?= <oss@mateuszmikula.dev>
+Date: Fri, 7 Feb 2025 22:33:39 +0100
+Subject: [PATCH 2/2] hack: add system search paths
+
+---
+ lld/MinGW/Driver.cpp | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/lld/MinGW/Driver.cpp b/lld/MinGW/Driver.cpp
+index 131bacdf1468..94acd68abd12 100644
+--- a/lld/MinGW/Driver.cpp
++++ b/lld/MinGW/Driver.cpp
+@@ -546,6 +546,10 @@ bool link(ArrayRef<const char *> argsArr, llvm::raw_ostream &stdoutOS,
+     add("-exclude-symbols:" + StringRef(a->getValue()));
+ 
+   std::vector<StringRef> searchPaths;
++  #ifdef __CYGWIN__
++    searchPaths.push_back("/usr/lib");
++    searchPaths.push_back("/usr/lib/w32api");
++  #endif
```

Comment:
This  breaks cross-compilation.  Shouldn't this be done in the compiler, anyway?

Reply:
IIRC, GCC doesn't pass `-L` so without this change `gcc -fuse-ld=lld` won't work.

Reply:
so this is another case where GNU ld hardcodes things because it knows its target at build time, but LLD is left to divine its target from its inputs, and there's no real way to tell the difference between mingw and cygwin from them at this point?

Reply:
This needs a double check but I think so.